pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,443 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package base
+
+import (
+	"encoding/json"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// ObjectType : Predefined object types
+type ObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownType  ObjectType = ""
+	LogType      ObjectType = "log"
+	RelationType ObjectType = "relation"
+)
+
+// LogObjectType :
+type LogObjectType string
+
+const (
+	// UnknownLogType : Invalid log type
+	UnknownLogType LogObjectType = ""
+	// ImageLogType :
+	ImageLogType LogObjectType = "image"
+	// NetworkInstanceLogType :
+	NetworkInstanceLogType LogObjectType = "network_instance"
+	// AppInstanceStatusLogType :
+	AppInstanceStatusLogType LogObjectType = "app_instance_status"
+	// AppInstanceConfigLogType :
+	AppInstanceConfigLogType LogObjectType = "app_instance_config"
+	// VolumeLogType :
+	VolumeLogType LogObjectType = "volume_type"
```

Comment:
We should decide whether we want one type for config+status (as you have for volume) or separate ones (as you have form appInstance)

Reply:
I think we should use different types for configuration and status.

Reply:
OK, so the introduce the volume config one since that's what you need for the relationship example.

I wonder whether we should derive the names using TypeToName. If we do we'll still have names (like "image") which are not derived from a type but it would help avoid typos if we derive the rest of the types.

Reply:
will introduce VolumeConfigLogType.

Users of this module can always fill in ```LogObjectType(TypeToName(myObject))``` of log object type.
I don't think we stop someone from doing that with current code. I believe it is simpler that way because we are not passed the external object (app instance etc) to determine it's type.